staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -600,6 +686,37 @@ func (p *sharedProcessor) addListenerLocked(listener *processorListener) {
 	p.syncingListeners = append(p.syncingListeners, listener)
 }
 
+func (p *sharedProcessor) removeListener(listener *processorListener) {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	if p.removeListenerLocked(listener) {
+		if p.listenersStarted {
+			close(listener.addCh)
+		}
+	}
+}
+
+func (p *sharedProcessor) removeListenerLocked(listener *processorListener) bool {
+	found := false
+	for i := 0; i < len(p.listeners); i++ {
+		l := p.listeners[i]
+		if l == listener {
+			p.listeners = append(p.listeners[:i], p.listeners[i+1:]...)
+			i--
+			found = true
```

Comment:
You can break here unless the listener can get in the loop twice? (another reason to switch to maps)

Reply:
break added and i-- removed. Basically the former coding was more defensive.